### PR TITLE
Re-enable progress on system tray icon

### PR DESCRIPTION
**Pull request by yavuzmert**
#577

### CODE REVIEW

File: src/core/qtsystemtrayicon.cpp [hunk 1]
```diff
@@ -54,7 +54,11 @@ QtSystemTrayIcon::QtSystemTrayIcon(QObject *parent)
       action_mute_(nullptr) {
 
   app_name_[0] = app_name_[0].toUpper();
-
+  QIcon theme_icon_grey =
```
Comment (jonaski) [inline src/core/qtsystemtrayicon.cpp:57]: One line please

File: src/core/qtsystemtrayicon.cpp [hunk 2]
```diff
@@ -54,7 +54,11 @@ QtSystemTrayIcon::QtSystemTrayIcon(QObject *parent)
...
+      IconLoader::Load("strawberry-grey");
+  if (! theme_icon_grey.isNull()) {
```
Comment (jonaski) [inline src/core/qtsystemtrayicon.cpp:59]: remove space after !

**Review by jonaski — CHANGES_REQUESTED** [review]
It would be nice if you could make a setting for this in the top options of the behavior settings (settings/appearancesettingspage)

**Comment by yavuzmert** [comment]
Hi Jonas, 
Setting and icon progress seem to work - at least on my system :)
I put the checkbox at the bottom of appearance settings, later i saw that you said top of the page. I think bottom is alright but you may reorganize it and put wherever you see appropriate.

File: src/core/systemtrayicon.cpp [hunk 3]
```diff
@@ -47,31 +49,36 @@ SystemTrayIcon::SystemTrayIcon(QObject *parent)
...
-
   QRect rect(icon.rect());
 
-  // The angle of the line that's used to cover the icon.
-  // Centered on rect.topRight()
-  double angle = double(100 - song_progress()) / 100.0 * M_PI_2 + M_PI;
-  double length = sqrt(pow(rect.width(), 2.0) + pow(rect.height(), 2.0));
+  QSettings s;
+  s.beginGroup(AppearanceSettingsPage::kSettingsGroup);
+  bool trayicon_progress = s.value(AppearanceSettingsPage::kTrayIconProgress, false).toBool();
```
Comment (jonaski) [inline src/core/systemtrayicon.cpp:56]: Move this to MainWindow::ReloadSettings() so we don't reload the setting each time the icon is painted.
Then add a public method in SystemTrayIcon to call when the setting is reloaded.

File: src/settings/appearancesettingspage.cpp [hunk 4]
```diff
@@ -197,6 +199,10 @@ void AppearanceSettingsPage::Load() {
   ui_->blur_slider->setValue(s.value(kBlurRadius, kDefaultBlurRadius).toInt());
   ui_->opacity_slider->setValue(s.value(kOpacityLevel, kDefaultOpacityLevel).toInt());
 
+#if !defined(Q_OS_WIN)
```
Comment (jonaski) [inline src/settings/appearancesettingspage.cpp:202]: Windows uses QSystemTrayIcon  too so this isn't needed.

**Review by jonaski — CHANGES_REQUESTED** [review]

**Comment by jonaski** [comment]
Sorry I meant behavior settings, settings/behavioursettingspage. I think it makes more sense since that's where the system tray icon options are.

**Comment by yavuzmert** [comment]
Alright I moved it to behavioursettingspage, I set it enabled when tray icon is checked. 
But only problem I see is, on MacOs, tray icon settings are hidden.
We can show its checkbox also on MacOs although "Show system tray icon" checkbox is hidden, but then we must set its label which makes sense all platforms (maybe "show progress on icon" instead of "show progress on system tray icon")

**Comment by jonaski** [comment]
The track progress doesn't work on macOS so you can just exclude that option. It's not really a system tray icon on mac at all, it's just the context menu on the dock.

File: src/core/mainwindow.cpp [hunk 5]
```diff
@@ -1032,8 +1032,11 @@ void MainWindow::ReloadSettings() {
 
   s.beginGroup(AppearanceSettingsPage::kSettingsGroup);
   int iconsize = s.value(AppearanceSettingsPage::kIconSizePlayControlButtons, 32).toInt();
+  bool trayicon_progress = s.value(AppearanceSettingsPage::kTrayIconProgress, false).toBool();
```
Comment (jonaski) [inline src/core/mainwindow.cpp:1035]: Move to behavioursettings

**Review by jonaski — CHANGES_REQUESTED** [review]

**Review by jonaski — APPROVED** [review]

**Pull request merged**